src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -1490,12 +1487,11 @@ void SystemDictionary::define_instance_class(InstanceKlass* k, Handle class_load
   }
 
   // Add the new class. We need recompile lock during update of CHA.
```

Comment:
This comment belongs after `add_to_hierarchy`.

Reply:
Not sure that make sense. Also we have no "recompile" lock.
I plainly removed this comment.

Reply:
"recompile" lock may have been a poor choice of words but the comment was describing why we need the Compile_lock when updating the dictionary.

Reply:
The "add class" I interpret was for { add_to_hierarchy() + update_dictionary() } combo.
I thought maybe there use to be a recompile ? :)

Let me know if I should add back something?